### PR TITLE
Handle libboost same as all other libraries

### PR DESCRIPTION
**Pull request by gtrensch**
This PR complements #1502. It implements the following:
- handle libboost same as all other 3rd-party libraries
- wrap libboost installation in extras/install_libboost.sh
- introduce travis environment variable xLIBBOOST to the build matrix
- exclude libboost from static code check
- move matplotlib configuration to the python section
- some minor formatting

### CODE REVIEW

File: extras/install_libboost.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Install the boost library
+wget --no-verbose https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.gz
+tar -xzf boost_1_72_0.tar.gz
+rm -fr boost_1_72_0.tar.gz
+cp -fr boost_1_72_0 $HOME/.cache
```
Comment (hakonsbm) [inline extras/install_libboost.sh]: Should it be installed to a directory `$HOME/.cache/boost.install` to make it consistent with the other libraries?

**Review by hakonsbm — CHANGES_REQUESTED** [review]
Thanks!

File: extras/install_libboost.sh (hunk 1 above)
Comment (gtrensch) [reply extras/install_libboost.sh]: @hakonsbm Yes, good point!

Comment (gtrensch) [reply extras/install_libboost.sh]: I have applied the change you suggested.
Seems that GitHub/Travis has problems to retrieve the changed files when I create a PR against your repo. The build ran successfully under my account on Travis: [Travis build](https://travis-ci.org/github/gtrensch/nest-simulator/builds/677304883)
I think it's safe if you merge the PR.

Comment (hakonsbm) [reply extras/install_libboost.sh]: Yes, looks like the static check doesn't handle PRs that are not against master that well.

**Review by hakonsbm — APPROVED** [review]

**Pull request merged**